src/pkg/leaderElection.go

```diff
@@ -0,0 +1,64 @@
+package pkg
+
+import (
+	"context"
+	"github.com/rs/zerolog/log"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"time"
+)
+
+var (
+	isLeader bool
+)
+
+func RunLeaderElection(lock *resourcelock.LeaseLock, ctx context.Context, id string) {
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Lock:            lock,
+		ReleaseOnCancel: true,
+		LeaseDuration:   15 * time.Second,
+		RenewDeadline:   10 * time.Second,
+		RetryPeriod:     2 * time.Second,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(c context.Context) {
+				isLeader = true
+				log.Info().Msgf("Perform Migration")
+				for {
+					log.Info().Msgf("leader is %s", id)
+					log.Info().Msgf("Getting replica count...")
```

Comment:
For the next ticket, this is where I was planning on inserting the get/set replica count for the deployment. #TODO